### PR TITLE
[WIP] EURO uasr scripts

### PR DESCRIPTION
**Pull request by DongjiGao**
Revert the previous modification to tokenize_text.py script:
- do tokenization only using tokenize_text
- get vocabulary in later in pyscripts/text/combine_text_and_vocab.py

### CODE REVIEW

**Comment by sw005320** [comment]
Thanks!
I just want to make sure the relationship between this PR and #4774.
Is it a part of #4774 or independent but necessary PR for #4774?

**Comment by ftshijt** [comment]
This is a part of #4774. This PR will be included in that PR after merge (to uasr branch)

File: espnet2/uasr/generator/conv_generator.py [hunk 1]
```diff
@@ -67,6 +67,7 @@ def __init__(
             self.pad = cfg.generator_pad
             self.bias = cfg.generator_bias
             self.dropout = torch.nn.Dropout(cfg.generator_dropout)
+            # TODO: fix batch_norm
```
Comment (ftshijt) [inline espnet2/uasr/generator/conv_generator.py:70]: Could you make it the format `TODO(your_name): comment` so that we can locate the issue with POC

File: espnet2/uasr/loss/smoothness_penalty.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    def forward(
+        self,
+        dense_logits: torch.Tensor,
+        dense_padding_mask: torch.Tensor,
+        sample_size: int,
+        is_discriminative_step: bool,
+    ):
+        """Forward.
+
+        Args:
```
Comment (ftshijt) [inline espnet2/uasr/loss/smoothness_penalty.py:34]: Given that this time, we aim for the full documentation. Could you also complete the comments here?

**Review by ftshijt — COMMENTED** [review]
Minor issues.

File: espnet2/uasr/generator/conv_generator.py (hunk 1 above)
Comment (DongjiGao) [reply espnet2/uasr/generator/conv_generator.py]: done

File: espnet2/uasr/loss/smoothness_penalty.py (hunk 2 above)
Comment (DongjiGao) [reply espnet2/uasr/loss/smoothness_penalty.py:34]: done

File: egs2/TEMPLATE/asr1/pyscripts/feats/apply_pca.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+#    copyfile(data_poth + ".lengths", save_path + ".lengths")
+
+#    if osp.exists(source_path + ".phn"):
+#        copyfile(source_path + ".phn", save_path + ".phn")
+#
+#    if osp.exists(source_path + ".wrd"):
+#        copyfile(source_path + ".wrd", save_path + ".wrd")
+#
+#    if osp.exists(save_path + ".npy"):
+#        os.remove(save_path + ".npy")
```
Comment (ftshijt) [inline egs2/TEMPLATE/asr1/pyscripts/feats/apply_pca.py:71]:
```suggestion
```

File: egs2/TEMPLATE/asr1/pyscripts/feats/apply_pca.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    return parser
+
+
+def main():
+    parser = get_parser()
+    args = parser.parse_args()
+
+#    source_path = osp.join(args.source, args.split)
+#    data_poth = source_path + "_unfiltered" if args.unfiltered else source_path
+
```
Comment (ftshijt) [inline egs2/TEMPLATE/asr1/pyscripts/feats/apply_pca.py:40]:
```suggestion
```

File: egs2/TEMPLATE/asr1/scripts/feats/feats_clustering.sh [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+set -u 
+set -o pipefail
+
+cmd=""
+nj=1
+dim=512
+num_clusters=128
+train_set="train"
+valid_set="valid"
+#test_set="test"
```
Comment (ftshijt) [inline egs2/TEMPLATE/asr1/scripts/feats/feats_clustering.sh:20]:
```suggestion
```

File: egs2/TEMPLATE/asr1/scripts/feats/feats_clustering.sh [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+#feats_scp="${uasr_stats_dir}/${test_set}/collect_feats/feats.scp"
+#split_dir="${uasr_stats_dir}/${test_set}/collect_feats/split${nj}"
+#mkdir -p "${split_dir}"
+#test_split_feats_scp=""
+#for n in $(seq ${nj}); do
+#    mkdir -p "${split_dir}/${n}"
+#    test_split_feats_scp="${test_split_feats_scp} ${split_dir}/${n}/feats.scp"        
+#done
+#utils/split_scp.pl "${feats_scp}" ${test_split_feats_scp}
```
Comment (ftshijt) [inline egs2/TEMPLATE/asr1/scripts/feats/feats_clustering.sh:72]:
```suggestion
```

File: egs2/TEMPLATE/asr1/scripts/feats/feats_clustering.sh [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Begin configuration section.
+set -e 
+set -u 
+set -o pipefail
+
+cmd=""
```
Comment (ftshijt) [inline egs2/TEMPLATE/asr1/scripts/feats/feats_clustering.sh:14]: can we set a default cmd?

File: egs2/TEMPLATE/asr1/pyscripts/feats/apply_pca.py (hunk 3 above)
Comment (DongjiGao) [reply egs2/TEMPLATE/asr1/pyscripts/feats/apply_pca.py:71]: deleted

File: egs2/TEMPLATE/asr1/pyscripts/feats/apply_pca.py (hunk 4 above)
Comment (DongjiGao) [reply egs2/TEMPLATE/asr1/pyscripts/feats/apply_pca.py:40]: deleted

File: egs2/TEMPLATE/asr1/scripts/feats/feats_clustering.sh (hunk 5 above)
Comment (DongjiGao) [reply egs2/TEMPLATE/asr1/scripts/feats/feats_clustering.sh:20]: done

File: egs2/TEMPLATE/asr1/scripts/feats/feats_clustering.sh (hunk 7 above)
Comment (DongjiGao) [reply egs2/TEMPLATE/asr1/scripts/feats/feats_clustering.sh:14]: run.pl?

Comment (ftshijt) [reply egs2/TEMPLATE/asr1/scripts/feats/feats_clustering.sh]: sounds good to me

**Comment by ftshijt** [comment]
Could you let me know if the PR is ready to merge? The PR LGTM! If the result is as expected, let's fix it and start to do the unit test/integration test/documentation.
